p8e-api/src/main/kotlin/io/provenance/engine/service/ChaincodeInvokeService.kt

```diff
@@ -285,15 +288,20 @@ class ChaincodeInvokeService(
     fun addContractSpecs(specList: List<ContractSpec>) {
         log.info("received a set of contract specs: ${specList.size}")
         try {
-            synchronized(sc) { // TODO - Replace after service account setup
-                batch {
-                    specList.map {
-                        prepare { sc.contractSpecs.addP8EContractSpec(it) }
-                    }
-                }.also {
-                    scBatchTx(it).also {
-                        log.info("batch made it to mempool with txhash = ${it.txhash}")
+            val txBody = specList.map {
+                MsgAddP8eContractSpecRequest.newBuilder()
+                    .setContractspec(it.toProv())
+                    .addAllSigners(listOf(accountProvider.bech32Address()))
+                    .build()
+            }.toTxBody()
```

Comment:
added this helper `toTxBody` to easily take a list of messages or an individual message and get the TxBody object with properly packed Any protos...

@scirner22 should be useful once you have a MsgP8eMemorializeContractRequest built